x-pack/plugins/enterprise_search/server/lib/enterprise_search_request_handler.ts

```diff
@@ -113,11 +118,17 @@ export class EnterpriseSearchRequestHandler {
           return this.handleInvalidDataError(response, url, json);
         }
 
+        // Intercept data that is meant for the server side session
+        const { _sessionData, ...responseJson } = json;
```

Comment:
There isn't a particular reason I went with a JSON key vs a header (other than on the server side setting a header on a Rails response is a little unusual...but then this whole interaction is unusual 😄 ).

I wouldn't use a cookie in this instance. Since our client (`node-fetch`) isn't storing cookies, using a cookie would be functionally the same as using a header without any of the persistence characteristics one would expect of a cookie.

If you would prefer a header (something like `x-ent-search-session-data`) I don't think it would be a very difficult change to make.

The only technical reason I can think to prefer something in the body over a header is that a header is effectively limited to 8K ([some web servers will choke on larger headers](https://stackoverflow.com/questions/686217/maximum-on-http-header-values)) whereas a response body doesn't really have a limit. However, we really shouldn't be trying to store anything that large in the session. The current use case shouldn't get even close to that.

Reply:
Sweet! Thanks for satisfying my curiosity!

Reply:
Also terrible skimming job on my part, just realized you had a question in here - I don't have a super strong feeling about `_sessionData` staying in the JSON body vs being a header. The only reason I can think of to prefer a header string is that you don't have to do any extra work splitting it out from the body JSON, but that's pretty minor and the unit test you've written for it satisfies any concerns I might have had there. Thanks again for the thoroughness here! 🎉